### PR TITLE
BAEL-2417 Added code samples for passing parameters to Java Thread.

### PR DESCRIPTION
**Pull request by sandy03934**

### CODE REVIEW

File: spring-data-rest/pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
             <artifactId>sqlite-jdbc</artifactId>
             <version>${sqlite.version}</version>
         </dependency>
-        <!-- <dependency> <groupId>org.hsqldb</groupId> <artifactId>hsqldb</artifactId> <version>${hsqldb.version}</version> </dependency> <dependency> <groupId>org.apache.derby</groupId> 
+        <!-- <dependency> <groupId>org.hsqldb</groupId> <artifactId>hsqldb</artifactId> <version>${hsqldb.version}</version> </dependency> <dependency> <groupId>org.apache.derby</groupId>
```
[the remark this answers is not in the document]
Comment (sandy03934) [reply spring-data-rest/pom.xml]: Sure!

File: core-java-concurrency/src/test/java/com/baeldung/parameters/ParameterisedThreadUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        } finally {
+            executorService.shutdown();
+        }
+    }
+
+    @Test
+    public void whenParametersToThreadWithLamda_thenParametersPassedCorrectly() throws Exception {
+        ExecutorService executorService = Executors.newFixedThreadPool(2);
+        int[] numbers = new int[] { 4, 5, 6 };
+        Future<Integer> sumResult = executorService.submit(sumCalculator(numbers));
```
[the remark this answers is not in the document]
Comment (sandy03934) [reply core-java-concurrency/src/test/java/com/baeldung/parameters/ParameterisedThreadUnitTest.java]: PR updated.

File: core-java-concurrency/src/test/java/com/baeldung/parameters/ParameterisedThreadUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.baeldung.concurrent.parameter.AverageCalculator;
+import org.junit.Test;
+
+import java.util.concurrent.*;
+
+import static com.baeldung.concurrent.parameter.ParameterizedThreadExample.*;
+import static org.junit.Assert.assertEquals;
+
+public class ParameterisedThreadUnitTest {
```
[the remark this answers is not in the document]
Comment (sandy03934) [reply core-java-concurrency/src/test/java/com/baeldung/parameters/ParameterisedThreadUnitTest.java]: Corrected.

**Comment by jzheaux** [comment]
@sandy03934 Note that you should rebase your local repository so that you don't get merges from past articles in the PR.

**Pull request merged**